src/agentic/actor_agents.py

```diff
@@ -1189,14 +1208,17 @@ def start_request(self, request: str, request_context: dict = {},
         if isinstance(request, str):
             request = self._check_for_prompt_match(request)
 
+        if thread_id is not None:
+            self._reload_thread_history(thread_id)
+
         if not thread_id and "thread_id" in request_context:
             thread_id = request_context["thread_id"]
 
         # Create request ID if not provided in continue_result
         request_id = continue_result.get("request_id") or str(uuid.uuid4())
 
         agent_instance = self._get_agent_for_request(request_id)
-        if (self.thread_id != thread_id or not self.thread_id) and self.db_path:
+        if thread_id is not None and (self.thread_id != thread_id or not self.thread_id) and self.db_path:
```

Comment:
without checking if thread_id is not None, then you end up getting new threads on every start_request.